### ISSUE TITLE
[Bug]: Unexpected change in no_data_state during atlantis apply: Expected "OK", but got "NoData"

### ISSUE THREAD

**Issue by shradha474** [grafana/terraform-provider-grafana#1843]
### Terraform Version

~> 1.3

### Terraform Grafana Provider Version

3.10.0

### Grafana Version

v11

### Affected Resource(s)

- grafana alert rules

### Terraform Configuration Files

```hcl
resource "grafana_machine_learning_alert" "anomaly_alerts" {
  for_each          = local.anomaly_alerts
  job_id            = grafana_machine_learning_job.anomaly_jobs[each.key].id
  title             = "${local.namespace}.${each.value.name}"
  anomaly_condition = each.value.anomaly_condition
  threshold         = each.value.threshold
  window            = each.value.window
  annotations       = each.value.annotations
  for               = each.value.pending_period
  labels            = each.value.labels
  no_data_state     = each.value.no_data_state
```

### Expected Behavior

When we update the `no_data_state` variable to `"OK"`, it should correctly change the value from `"NoData"` to `"OK"` to prevent triggering anomaly alerts during periods of no data.

### Actual Behavior

When we attempt to update the `no_data_state` variable to "OK", the plan reflects the expected change. However, the apply fails with an error. 

```
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to
│ grafana_machine_learning_alert.anomaly_alerts["anomaly_testing_configuration-p95-latency-anomaly"],
│ provider "provider[\"registry.terraform.io/grafana/grafana\"]" produced an
│ unexpected new value: .no_data_state: was cty.StringVal("OK"), but now
│ cty.StringVal("NoData").
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
```

### Steps to Reproduce

1. Update the no_data_state value to "OK"
2. Atlantis apply

**Comment by spinillos** [comment]
Can you try to update to the latest version of terraform provider and try it again?

**Comment by shradha474** [comment]
@spinillos , we are using the latest `3.10.0` terraform provider version and getting the same error.

**Comment by shradha474** [comment]
This issue has been resolved in version `3.10.0`.
